.github/actions/csv-vulnerability-filter/src/processing.ts

```diff
@@ -0,0 +1,341 @@
+}
+
+export function isFixedVersionPresent(row: CsvRow): boolean {
+  const fixedVer = row['Fixed Package'];
+  if (!fixedVer) return false;
+  const trimmed = fixedVer.trim().toUpperCase();
+  return trimmed !== '' && trimmed !== 'N/A' && trimmed !== 'NONE' && !trimmed.startsWith('NO_FIX');
+}
+
+export function generateCsvString(data: CsvRow[]): string {
+  if (data.length === 0) {
+    return '';
+  }
+  try {
+    const csvString = Papa.unparse(data, {
+      header: true,
+    });
+    return csvString;
+  } catch (error) {
+    core.error(`Failed to generate CSV string: ${error instanceof Error ? error.message : String(error)}`);
+    throw new Error('CSV string generation failed.');
+  }
+}
+
+export function writeOutputCsv(csvString: string, baseFilename: string): string {
+  const tempDir = process.env.RUNNER_TEMP;
+  if (!tempDir) {
+    core.warning("RUNNER_TEMP environment variable not defined. Using current directory for output.");
+    const fallbackPath = path.resolve('.');
+    const uniqueId = Math.random().toString(36).substring(2, 8);
+    const filename = `${baseFilename}_${uniqueId}.csv`;
+    const outputPath = path.join(fallbackPath, filename);
+    core.info(`Writing output CSV to fallback location: ${outputPath}`);
+    try {
+      fs.writeFileSync(outputPath, csvString, 'utf8');
+      return outputPath;
+    } catch (error) {
+      core.error(`Failed to write output CSV file ${outputPath}: ${error instanceof Error ? error.message : String(error)}`);
+      throw new Error(`Failed to write output CSV to fallback location: ${outputPath}`);
+    }
+  }
+
+  const uniqueId = Math.random().toString(36).substring(2, 8);
+  const filename = `${baseFilename}_${uniqueId}.csv`;
+  const outputPath = path.join(tempDir, filename);
+
+  core.info(`Writing output CSV to: ${outputPath}`);
+  try {
+    fs.writeFileSync(outputPath, csvString, 'utf8');
+    return outputPath;
+  } catch (error) {
+    core.error(`Failed to write output CSV file ${outputPath}: ${error instanceof Error ? error.message : String(error)}`);
+    throw new Error(`Failed to write output CSV: ${outputPath}`);
+  }
+}
+
+export function writeOutputMarkdown(markdownString: string, baseFilename: string): string {
+  const tempDir = process.env.RUNNER_TEMP;
+  let outputPath: string;
+
+  if (!tempDir) {
+    core.warning("RUNNER_TEMP environment variable not defined. Using current directory for Markdown output.");
+    const fallbackPath = path.resolve('.');
+    const uniqueId = Math.random().toString(36).substring(2, 8);
+    const filename = `${baseFilename}_${uniqueId}.md`;
+    outputPath = path.join(fallbackPath, filename);
+    core.info(`Writing output Markdown to fallback location: ${outputPath}`);
+  } else {
+    const uniqueId = Math.random().toString(36).substring(2, 8);
+    const filename = `${baseFilename}_${uniqueId}.md`;
+    outputPath = path.join(tempDir, filename);
+    core.info(`Writing output Markdown to: ${outputPath}`);
+  }
```

Comment:
Some repeated code here, could be consolidated to something like:
```suggestion
  let outputDir = process.env.RUNNER_TEMP;
  const usingFallback = !!outputDir;
  if (!outputDir) {
    core.warning("RUNNER_TEMP environment variable not defined. Using current directory for Markdown output.");
    outputDir = path.resolve('.');
  }
  const uniqueId = Math.random().toString(36).substring(2, 8);
  const filename = `${baseFilename}_${uniqueId}.md`;
  const outputPath = path.join(outputDir, filename);
  core.info(`Writing output Markdown to${usingFallback ? ' fallback location' : ''}: ${outputPath}`);
```